prefect_gcp/credentials.py

```diff
@@ -86,22 +87,30 @@ class GcpCredentials(Block):
     service_account_file: Optional[Path] = None
     service_account_info: Optional[Union[Dict[str, str], Json]] = None
     project: Optional[str] = None
+    infer_project: bool = False
```

Comment:
I think that we need to be careful about adding this new attribute since it will show up in the UI and might be confusing for users. I think the way that we infer the project already works and won't fix the errors that we're seeing in prefect-dbt. I'm of the opinion that those error should be addressed in prefect-dbt and not here.